Feature/slurmdbd

hfrSchmidt opened the pull request:
First part of incorporating slurmdbd in the slurm setup.

hfrSchmidt committed:
Adapt slurmctld image to build slurmdbd

This builds a slurmdbd image using the same methodology as the slurmctld
image.

hfrSchmidt committed:
Add gh action to build the slurmdbd image

enasca reviewed (changes requested):
Let's see if we can turn this into a 2-stage build with a base Dockerfile for the shared elements, then 2 Dockerfiles on top of it: one for slurmctld, one for slurmdbd.

hfrSchmidt committed:
Rework Docker build setup to use a common base image

hfrSchmidt committed:
Merge branch 'main' into feature/slurmdbd

hfrSchmidt committed:
Add workflow for building the slurm base image

hfrSchmidt committed:
Setup docker buildx first

hfrSchmidt committed:
Remove on push and on pr action triggers

This does not make sense for the base image, since it is not really
useful on its own.

hfrSchmidt committed:
Use the base image to build the slurmctld image

hfrSchmidt committed:
Simplify and correct slurm base image

hfrSchmidt committed:
Derive slurmctld from the previously build base img

hfrSchmidt committed:
Derive slurmdbd from the slurm base img

hfrSchmidt committed:
Add the build of the base img as a reusable action

hfrSchmidt committed:
Derive the slurmdbd image from the base image

hfrSchmidt committed:
Fix base image workflow path

hfrSchmidt committed:
Change path to base image action to local path

hfrSchmidt committed:
Switch to using outputs in the base img generation

Makes it easier to retrieve the actual identifier of the base image

hfrSchmidt committed:
Remove output from manual trigger

hfrSchmidt committed:
Use the standard github method of passing outputs

hfrSchmidt committed:
Fix missing part of base img ref output path

hfrSchmidt committed:
Remove quotation marks around env var value

Due to the yml multiline string this is interpreted as a literal
quotation mark.

enasca commented:
Hopefully we can build both docker images using one workflow file. Also, uploading the base image to Github packages could make the workflow a bit easier.

hfrSchmidt committed:
Add ssh keys via a volume

hfrSchmidt committed:
Use a non-root user for the slurm daemons and mount ssh keys

hfrSchmidt committed:
Adding a volume is not necessary

The volume is not needed when later mounting the keys via a k8s
configmap into the pod.

hfrSchmidt committed:
Ensure that the /run/munge dir is present

hfrSchmidt committed:
Push the base image as well

hfrSchmidt committed:
Make building the base image a standalone action

hfrSchmidt committed:
Need to log into the registry to be able to push

hfrSchmidt committed:
Integrate both slurm components into one workflow

hfrSchmidt committed:
Remove obsolete workflow file

hfrSchmidt committed:
Fix context variable in action

hfrSchmidt committed:
Merge branch 'main' into feature/slurmdbd

hfrSchmidt committed:
Fix base image URL

hfrSchmidt committed:
Use shell expansion to convert repo to lowercase

hfrSchmidt committed:
Use shell expansion as part of a separate step

hfrSchmidt committed:
Fix BASE_IMG reference

hfrSchmidt committed:
Fix the tag of the base image to use

hfrSchmidt committed:
Rename the slurm container action and job

hfrSchmidt committed:
Unify slurm build action

hfrSchmidt committed:
Minor refactoring of names

hfrSchmidt committed:
Remove obsolete base image workflow

hfrSchmidt committed:
Use a suffix with the ubuntu version for the base image

hfrSchmidt committed:
Refactor base tag matrix variable name
